src/map/pc.cpp

```diff
@@ -1972,6 +1972,47 @@ bool pc_set_hate_mob(struct map_session_data *sd, int pos, struct block_list *bl
 	return true;
 }
 
+TIMER_FUNC(pc_goldpc_update){
+	struct map_session_data* sd = map_id2sd( id );
+
+	if( sd == nullptr ){
+		return 0;
+	}
+
+	sd->goldpc_tid = INVALID_TIMER;
```

Comment:
The timer should be removed.

	// Delete previous timer.
	if (sd->goldpc_tid != INVALID_TIMER) {
		delete_timer( sd->goldpc_tid, pc_goldpc_update );
		sd->goldpc_tid = INVALID_TIMER;
	}

Reply:
Why do you think that this is necessary?
All calls to this function should come from timer.cpp::do_timer and that auto-removes timers after executing the function attached to the timer.